guidelines/monai-application-package.md

```diff
@@ -1,498 +1,588 @@
+    - Documentation
 
-| Variable            | Default                                    | Format              | Description                                      |
-| ------------------- | ------------------------------------------ | ------------------- | ------------------------------------------------ |
-| `MONAI_APPLICATION` | `/etc/monai/pkg.json#application.location` | Folder Path         | Path to the Application.                         |
-| `MONAI_HOSTNAME`    | `None`                                     | String              | Name of the host running the MAP.                |
-| `MONAI_HOSTVERSION` | `0.0.0`                                    | Semantic Version    | Version of the host running the MAP.             |
-| `MONAI_INPUTPATH`   | `/var/monai/input/`                        | Folder Path         | Path to the input folder for the Application.    |
-| `MONAI_JOBID`       | `00000000000000000000000000000000`         | `/[A-F0-9a-f]{32}/` | Unique identity of the job the MAP is executing. |
-| `MONAI_OUTPUTPATH`  | `/var/monai/output/`                       | Folder Path         | Path to the output folder for the Application.   |
-| `MONAI_TIMEOUT`     | `/etc/monai/app.json#timeout`              | Integer             | Timeout, in seconds, applied to the Application. |
-| `MONAI_WORKDIR`     | `/var/monai/`                              | Folder Path         | Path to the Application's working directory.     |
-| `MONAI_MODELPATH`   | `/opt/monai/models/`                       | Folder Path         | Path to the Application's models directory.     |
+    - Third-party licenses
 
+### Container Behavior and Interaction
 
-### Manifest Export
+A MAP is a single container supporting the following defined behaviors when started.
 
-The Executor is able to function in a special mode in which it will export the MAP's manifest files to a mounted folder.
-This enables external tooling and services to read the manifest without any special tooling beyond the ability to run the MAP correctly.
+#### Default Behavior
 
-The Executor MUST detect when the following folders have been mounted.
+When a MAP is started from the CLI or other means without any parameters, the MAP shall execute the contained application. The MAP internally may use `Entrypoint`, `CMD`, or a combination of both.
 
-- `/var/run/monai/export/app/`: when detected, the Executor will copy the contents of `/opt/monai/app/` to the folder.
+#### Manifest Export
 
-- `/var/run/monai/export/config/`: when detected, the Executor will copy `/etc/monai/app.json` and `/etc/monai/pkg.json` to the folder.
+A MAP SHALL provide at least one method to access the _embedded application_, _models_, _licenses_, _README_, or _manifest files_, namely, `app.json` and `package.json`.
 
-- `/var/run/monai/export/models/`: when detected, the Executor will copy the contents of `/opt/monai/models/` to the folder.
+- The Method SHALL have the option to print one or more manifest files to the console.
```

Comment:
Since a `Specification` is defined here, and specifically the package is a container image, it is better to provide the exact expected `option` or argument.

Reply:
As you have suggested previously, this is a `Specification` and not to explicitly or restrict the users from using ENTRYPOINT, CMD. Therefore, we suggest the users to provide a method to print the manifest and how it is implemented is to be documented on their end.

Reply:
For "CLI", command and options are in the specifications, but how that option is supported/implemented is not the concern of the spec.

Please look at the use case for having this particular spec. A hosting app needs to extract the manifest files at runtime. If given a MAP which has a unknown `Method` of extracting the files, how would that work? If it has to run the container and use the WELL-KNOWN DEFAULT folders to get and copy the files out, what is use of the whole thing on `export`? 

As I showed in the App SDK [discussion](https://github.com/Project-MONAI/monai-deploy-app-sdk/discussions/340), it is trivial to script the task to get the files even without running the container, see below (and sub out the vars with manifest file paths in the container)
```
    echo "Extracting input DICOM files to ${INPUT_DCM}...."
    extractor=$(docker create $MAP)
    docker cp ${extractor}:${DCM_IN_MAP} ${INPUT_DCM}
    docker rm ${extractor}
    echo "Extracted input DICOM files in folder ${INPUT_DCM}:"
    ls ${INPUT_DCM}
```

Reply:
> For "CLI", command and options are in the specifications, but how that option is supported/implemented is not the concern of the spec.

I'm very confused; as you've mentioned, this is a spec, but your 2nd paragraph wants to define a `Method` explicitly.  We have generalized this statement per your **previous** feedback, so it does **NOT** explicitly define it since this is a **spec**.

Also, for hosting services, given that we spell out where `app.json` and `pkg.json` files are, they can always mount and copy the files without using this `Method`; this `Method` is really for the users.  Rahul have also suggested that for users, we will assume that the MAP publishers would include user manuals with instructions on how to use that `Method` for extracting data.

Reply:
The "Spec" contains the content structure and API definition (meaning the exact method names and behavior). I don't see what and where confusions were created by my statement.

The example showing how to get to the file content in the image merely shows practically OCI container and Docker already has the well known ways for one to get to the content. If the MAP spec does not provide any abstraction over it, what is the point of saying the MAP will have a method to do it?

Reply:
The "Spec" contains the content structure and API definition (meaning the exact method names and behavior). I don't see what and where confusions were created by my statement.

The example showing how to get to the file content in the image merely shows practically OCI container and Docker already has the well known ways for one to get to the content. If the MAP spec does not provide any abstraction over it, what is the point of saying the MAP will have a method to do it?

Reply:
Are you suggesting bringing back the SHIM or?

Reply:
No SHIM as it is one of the implementations to support the defined commands expected on a MAP. Updated the doc post discussion.